### DISCUSSION TITLE
Disable `mini.indentscope` in `vim.lsp.buf.hover` floating window

### DISCUSSION THREAD

**Discussion by FrancescoLuzzi** [nvim-mini/mini.nvim#962] [Q&A] [answered]
Hi everyone,
I was able to disable `indentscope` for some filetypes with:
```lua
    local disabled_filetypes = { 'NvimTree', 'alpha', 'help' }
    vim.api.nvim_create_autocmd('FileType', {
      group = vim.api.nvim_create_augroup('MiniIndentScopeDisable', { clear = true }),
      callback = function(opts)
        local ftype = vim.bo[opts.buf].filetype
        if vim.tbl_contains(disabled_filetypes, ftype) then
          vim.b[opts.buf].miniindentscope_disable = true
        end
      end,
    })
```

How would you go disabling `indentscope` inside the floating window opened by `vim.lsp.buf.hover`?
I've tried attaching an autocmd to `WinNew` (following this [comment](https://github.com/neovim/neovim/issues/26548#issuecomment-1852958452)) I just can't figure it out 😅

Thanks!

**Comment by echasnovski** [comment, marked as answer]
> How would you go disabling `indentscope` inside the floating window opened by `vim.lsp.buf.hover`?

Good question. Unfortunately, I also can't see a reliable way to specifically identify a buffer used in `vim.lsp.buf.hover()`.

Instead I can suggest a more general criteria which should fit nicely in what you're trying to achieve: disable 'mini.indentscope' in any "helper" buffer (i.e. with 'buftype' not equal to `''`) in floating window (i.e. with `relative` config field not equal to `''`). Here is how I'd go about implementing it:

```lua
local disable_in_helper_floats = function(opts)
  if vim.bo[opts.buf].buftype == '' then return end
  if vim.api.nvim_win_get_config(0).relative == '' then return end
  vim.b[opts.buf].miniindentscope_disable = true
end
local augroup = vim.api.nvim_create_augroup('MiniIndentScopeDisable', { clear = true })
vim.api.nvim_create_autocmd('BufWinEnter', { group = augroup, callback = disable_in_helper_floats })
```

**Reply by FrancescoLuzzi** [reply]
Worked like a charm, thank you 🚀
